### PR TITLE
PE-379 Set proper layout to ImageView and DrawingView

### LINKED ISSUE #379: Should disallow drawing on left or right of the image
**Describe the bug**
Currently, it is possible to draw on the outside (right and left only) of the selected image, due to the way the layout is set.

**To Reproduce**
1. Select an image that lets right and left sides empty (black on the demo app ;
2. Brush on these parts ;
3. You can see shapes on the outside of the image.

**Expected behavior**
It is not possible to draw outside of the given image, drawing should be cropped.

**Screenshot**
Screenshot_1626709982

**Smartphone Info:**
 - Version 1.5.0 on any device

### PR DESCRIPTION
**Pull request by stephanepechard**
in order not to draw outside of the source image. Here is the result with the same type of drawing as in the issue.

Screenshot_1626710746

Resolves #379

### CODE REVIEW

**Review by AtulYadav01 — APPROVED** [review]

**Comment by burhanrashid52** [comment]
There is an issue with CI. I am looking into it.

**Comment by stephanepechard** [comment]
As requested, I added a parameter to `PhotoEditor.Builder` to be able to switch this off, which is the default configuration.
I had to refactor a bit the `PhotoEditorView` `init()` method to ease the appliance of the new LayoutParams.

File: photoeditor/src/main/java/ja/burhanrashid52/photoeditor/PhotoEditor.java [hunk 1]
```diff
@@ -358,6 +358,16 @@ public Builder setPinchTextScalable(boolean isTextPinchScalable) {
...
             return new PhotoEditorImpl(this);
         }
+
+        /**
+         * Set true true to clip the drawing brush to the source image.
+         *
+         * @param clip a boolean to indicate if brush drawing is clipped or not.
+         */
+        public Builder setClipSourceImage(boolean clip) {
+            parentView.setClipSourceImage(clip);
```
Comment (burhanrashid52) [inline photoeditor/src/main/java/ja/burhanrashid52/photoeditor/PhotoEditor.java:368]: Why setting it from `Builder` ? This should be done when inside `PhotoEditorImpl` constructor when `build()` method is called. What I am missing here ?

Comment (stephanepechard) [reply photoeditor/src/main/java/ja/burhanrashid52/photoeditor/PhotoEditor.java]: You're right, it's cleaner this way. It's done now.

**Comment by burhanrashid52** [comment]
@stephanepechard Is it possible for you to create the PR for wiki documentation as well ?

File: photoeditor/src/main/java/ja/burhanrashid52/photoeditor/PhotoEditorView.java [hunk 2]
```diff
@@ -57,15 +58,45 @@ public PhotoEditorView(Context context, AttributeSet attrs, int defStyleAttr, in
...
+
+
+    @SuppressLint("Recycle")
+    private RelativeLayout.LayoutParams setupImageSource(@Nullable AttributeSet attrs) {
         mImgSource.setId(imgSrcId);
         mImgSource.setAdjustViewBounds(true);
-        RelativeLayout.LayoutParams imgSrcParam = new RelativeLayout.LayoutParams(
-                ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
-        imgSrcParam.addRule(RelativeLayout.CENTER_IN_PARENT, RelativeLayout.TRUE);
+        mImgSource.setScaleType(ImageView.ScaleType.CENTER_INSIDE);
```
Comment (burhanrashid52) [inline photoeditor/src/main/java/ja/burhanrashid52/photoeditor/PhotoEditorView.java:98]: Why change the scale type to CENTER_INSIDE ?

**Comment by burhanrashid52** [comment]
It is possible for you update the wiki and create a PR ? https://github.com/burhanrashid52/PhotoEditor.wiki.git

**Comment by stephanepechard** [comment]
Will do it, I just do not have much time at the moment, I'm in holidays :-)

**Comment by stephanepechard** [comment]
> It is possible for you update the wiki and create a PR ? https://github.com/burhanrashid52/PhotoEditor.wiki.git

I cannot fork this repo. How can I create a PR for this?

**Comment by burhanrashid52** [comment]
Yeah, we cannot do that. Never mind I will update the wiki. Enjoy your holidays!!! :)

**Comment by stephanepechard** [comment]
Thanks!

**Review by burhanrashid52 — APPROVED** [review]

**Pull request merged**